spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -129,10 +132,11 @@ protected boolean isReceiveOnlyAdvice(Advice advice) {
 	@Override
 	protected void applyReceiveOnlyAdviceChain(Collection<Advice> chain) {
 		if (AopUtils.isAopProxy(this.source)) {
+			this.appliedAdvices.stream().forEach(((Advised) this.source)::removeAdvice);
```

Comment:
The `forEach()` is a part of `Collection`. No reason in `.stream()`.
Plus here is an IDEA tip :smile:  : 
![foreachwithoutstream](https://cloud.githubusercontent.com/assets/1071590/18840869/00e44d88-83de-11e6-9aa0-1a26650315ff.png)